config/routes.rb

```diff
@@ -177,6 +177,8 @@
       resources :watched_words, only: [:index, :create, :update, :destroy] do
         collection do
           get "action/:id" => "watched_words#index"
+          get "action/:id/download" => "watched_words#download"
```

Comment:
Since we have an hardcoded limit of 1000 watched words per action, they all are serialized to the client.

I don't think we need to call the server to generate a CSV when we have all the information already available in the browser.

Reply:
We need to make sure to escape/quote the CSV correctly, in case the watched words contain commas. In Ruby we can do that easily with [CSV](https://ruby-doc.org/stdlib-2.6.1/libdoc/csv/rdoc/CSV.html). I'm not sure it's so easy in Javascript 🤔

Reply:
Actually I'm not sure why we require a CSV file here. When you upload a file, it's treated as a regular txt file and each line is treated as one word even though we require a CSV file:

https://github.com/discourse/discourse/blob/514aaacdf486613bda49f8c729867926fbceb174/app/controllers/admin/watched_words_controller.rb#L31-L33

Do you think there would be problems if we dropped the CSV requirement and switched to txt?

Reply:
In most cases, I think it would be fine. However, some people will be using other applications to generate the list. For example,  in excel I can add two words to a list and end up with this CSV:

```
"word,with,comma"
"word
with
newline"
```
Anyway, as long as we're consistent with imports and exports I don't think it matters. We should import/export `.csv` **or** import/export `.txt`

Reply:
OK I went with importing/exporting txt files.

@ZogStriP I tried to do the downloading on the client side, it worked well on chrom but didn't work on IE11 and Edge didn't work.